src/model/PoolState.ts

```diff
@@ -0,0 +1,38 @@
+import { ConfigurableCorePool } from "../core/ConfigurableCorePool";
+import { PoolConfig } from "../entity/PoolConfig";
+import { Record } from "../entity/Record";
+import { Snapshot } from "../entity/Snapshot";
+import { DBManager } from "../manager/DBManager";
+
+export class PoolState {
+  id: string = "0";
+  baseSnapshot: Snapshot | null;
+  snapshot: Snapshot | null = null;
+  poolConfig: PoolConfig;
+  records: Record[] = new Array<Record>();
+  timestamp: Date = new Date();
+
+  constructor(baseSnapshot: Snapshot | null = null, poolConfig: PoolConfig) {
```

Comment:
Consider using the `optional parameters`, see if it fits the need.

Reply:
I see this point. And actually here I want to choose one of two params as the input to construct the instance. I wonder if there is some good idea.

Reply:
I see, which one do you want to use with higher priority?
I think if both are on the same level of priority, it's actually better just to have 2 constructors, one for each, instead of a `if` switch, constructor apis like this are a little confusing.

Reply:
Actually typescript doesn't support multiple constructors and it supports constructor overloading like multiple signatures and one implementation. But I think (param1? : type1, param2? : type2) will work. Please have a look after next commit.